test/plugin/test_out_redis.rb

```diff
@@ -37,18 +49,113 @@ def test_configure_with_password
     assert_equal 'testpass', d.instance.password
   end
 
+  def test_configure_without_tag_chunk_key
+    config = config_element('ROOT', '', {
+                              "host" => "localhost",
+                              "port" =>  6379,
+                              "db_number" => 1,
+                            }, [
+                              config_element('buffer', 'time', {
+                                               'chunk_keys' => 'time',
+                                             })
+                            ])
+    assert_raise Fluent::ConfigError do
+      create_driver(config)
+    end
+  end
+
   def test_format
-    @d.emit({"a"=>1}, @time)
-    @d.expect_format(["test.#{@time}", {"a"=>1}].to_msgpack)
-    @d.run
+    @d.run(default_tag: 'test') do
+      @d.feed(@time, {"a"=>1})
+    end
+    assert_equal [["test", @time, {"a"=>1}].to_msgpack], @d.formatted
+  end
+
+  class InjectTest < self
+    def test_format_inject_tag_keys
+      d = create_driver CONFIG + %[
+        include_tag_key true
+      ]
+      d.run(default_tag: 'test') do
+        d.feed(@time, {"a"=>1})
+      end
+      assert_equal [["test", @time, {"a"=>1, "tag" => "test"}].to_msgpack], d.formatted
+    end
+
+    def test_format_inject_time_keys
+      d = create_driver CONFIG + %[
+        include_time_key true
+      ]
+      timef = time_formatter(d.instance.inject_config)
+      d.run(default_tag: 'test') do
+        d.feed(@time, {"a"=>1})
+      end
+      assert_equal [["test", @time, {"a"=>1, "time" => timef.call(@time)}].to_msgpack], d.formatted
+    end
   end
 
-  def test_write
-    @d.emit({"a"=>2}, @time)
-    @d.emit({"a"=>3}, @time)
-    @d.run
+  class WriteTest < self
+    def setup
+      Timecop.freeze(Time.parse("2011-01-02 13:14:00 UTC"))
+    end
+
+    def test_write
+      d = create_driver
+      time = Fluent::Engine.now
```

Comment:
> We should use Fluent::Engine.now instead of event_time, right?

`event_time` is a short hand for `Fluent::EventTime.now`.
I think we should use `event_time` in test code.